connection.go

```diff
@@ -163,6 +160,16 @@ func (conn *Connection) tx() error {
 				return err
 			}
 			log.Println(err)
+
+			// reduce window size
+			conn.Lock()
+			conn.windowSize /= 2
```

Comment:
Use setWindowSize() to avoid repeated code

Reply:
Yes, modified